### DISCUSSION TITLE
Exact GP Regression on Classification Labels with Dataloader

### DISCUSSION THREAD

**Discussion by giamp66** [cornellius-gp/gpytorch#2343] [Q&A] [answered]
How can I make an Exact GP Regression on Classification Labels with Dataloader ?
A little example would be greatly appreciated. 

I have tried mimiching dataloader used in "Stochastic Variational GP Regression" but i get an erro saying " You must train on the training inputs!"
Thanks.

**Comment by Turakar** [comment, marked as answer]
I am unsure what you are trying to achieve. A fundamental drawback of exact GPs is the requirement to train on the whole dataset in every step, as you need to compute the solve of the full covariance matrix with the expected output. If you want mini-batches, you have to use a variational method. However, there are ways to scale exact GPs to very large datasets. For this, have a look at the docs.
